chat/prekey/key_test.go

```diff
@@ -0,0 +1,74 @@
+package prekey
+
+import (
+	"encoding/hex"
+	keyManager "github.com/Bit-Nation/panthalassa/keyManager"
+	keyStore "github.com/Bit-Nation/panthalassa/keyStore"
+	mnemonic "github.com/Bit-Nation/panthalassa/mnemonic"
+	require "github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+func TestHashInvalidIdPublicKey(t *testing.T) {
+	k := PreKey{}
+	_, err := k.hash()
+	require.EqualError(t, err, "got invalid identity key public key")
+}
+
+func TestHashInvalidPublicKey(t *testing.T) {
+	k := PreKey{}
+	k.identityPublicKey = [32]byte{1}
+	_, err := k.hash()
+	require.EqualError(t, err, "got invalid pre key public key")
+}
+
+func TestSign(t *testing.T) {
+
+	mne, err := mnemonic.New()
+	require.Nil(t, err)
+
+	ks, err := keyStore.NewFromMnemonic(mne)
+	require.Nil(t, err)
+
+	km := keyManager.CreateFromKeyStore(ks)
+
+	k := PreKey{}
+	k.PublicKey = [32]byte{1}
+
+	require.Nil(t, k.Sign(*km))
+
+	valid, err := k.VerifySignature()
+	require.Nil(t, err)
+	require.True(t, valid)
+
+}
+
+func TestPreKey_ToProtobufAndBack(t *testing.T) {
+
+	now := time.Now()
```

Comment:
Might be preferable to use time.Now().UnixNano() instead of time.Now() as UnixNano leaves almost no room for error if we ever need to covert to different timezones or time formats.